app/models/assignment.rb

```diff
@@ -688,26 +689,32 @@ def criterion_class
   end
 
   # Returns a filtered list of criteria.
-  def get_criteria(user_visibility = :all)
+  def get_criteria(user_visibility = :all, type = :all)
     if user_visibility == :all
-      get_all_criteria
+      get_all_criteria(type)
     elsif user_visibility == :ta
-      get_ta_visible_criteria
+      get_ta_visible_criteria(type)
     elsif user_visibility == :peer
-      get_peer_visible_criteria
+      get_peer_visible_criteria(type)
     end
   end
 
-  def get_all_criteria
-    criterion_class.where(assignment_id: id).order(:position)
+  def get_all_criteria(type)
+    if type == :all
+      (RubricCriterion.where(assignment_id: id) + FlexibleCriterion.where(assignment_id: id)).sort_by(&:position)
```

Comment:
You don't need to use `where` here. Assignment already has an association for both rubric and flexible criterion. (Should have caught this one earlier, but oh well.)